app/blog/[...slug]/page.tsx

```diff
@@ -1,53 +1,56 @@
-import { ReactNode } from "react";
+import fs from "fs";
+
+import { Metadata } from "next";
+import { notFound } from "next/navigation";
 
 import Link from "@/components/Link";
 import PageTitle from "@/components/PageTitle";
 import SectionContainer from "@/components/SectionContainer";
-import { BlogSEO } from "@/components/SEO";
 import Image from "@/components/Image";
-import Tag from "@/components/Tag";
-import siteMetadata from "@/data/siteMetadata";
 import ScrollTop from "@/components/ScrollTop";
-import { AuthorFrontMatter, PostFrontMatter } from "types/FrontMatter";
+import MdxRenderer from "@/components/MdxComponents";
+import siteMetadata from "@/data/siteMetadata";
 
-const editUrl = (fileName) =>
-  `${siteMetadata.siteRepo}/blob/master/data/blog/${fileName}`;
-const discussUrl = (slug) =>
-  `https://mobile.twitter.com/search?q=${encodeURIComponent(
-    `${siteMetadata.siteUrl}/blog/${slug}`,
-  )}`;
+import { sortedBlogs, authors } from "@/data/generated";
+import generateRss from "@/lib/generate-rss";
 
+// "Saturday, July 1st, 2023"
 const postDateTemplate: Intl.DateTimeFormatOptions = {
   weekday: "long",
   year: "numeric",
   month: "long",
   day: "numeric",
 };
 
-interface PostLayoutProps {
-  frontMatter: PostFrontMatter;
-  authorDetails: AuthorFrontMatter[];
-  next?: { slug: string; title: string };
-  prev?: { slug: string; title: string };
-  children: ReactNode;
+interface PostProps {
+  params: { slug: string[] };
 }
 
-const PostLayout = ({
-  frontMatter,
-  authorDetails,
-  next,
-  prev,
-  children,
-}: PostLayoutProps) => {
-  const { slug, fileName, date, title, tags } = frontMatter;
+const Post = ({ params }: PostProps) => {
+  const slug = params.slug.join("/");
+
+  const postIndex = sortedBlogs.findIndex((p) => p.slug === slug);
+
+  if (postIndex < 0) notFound();
+
+  const post = sortedBlogs[postIndex];
+  const postAuthors = post.authors || [siteMetadata.author];
+  const authorDetails = postAuthors.map((authorName) => authors[authorName]);
+
+  const { filePath, date, title, draft } = post;
+
+  if (draft) notFound();
+
+  const prevPost = sortedBlogs[postIndex + 1] || null;
+  const nextPost = sortedBlogs[postIndex - 1] || null;
 
   return (
     <SectionContainer>
-      <BlogSEO
-        url={`${siteMetadata.siteUrl}/blog/${slug}`}
-        authorDetails={authorDetails}
-        {...frontMatter}
-      />
+      {/*<BlogSEO*/}
```

Comment:
Can be removed, added `generateMetadata` below. Can remove the entire `BlogSEO` component, and hopefully the rest of the SEO components as well.